src/cargo/ops/lockfile.rs

```diff
@@ -43,6 +43,15 @@ pub fn write_pkg_lockfile(ws: &Workspace, resolve: &Resolve) -> CargoResult<()>
 
     let mut out = String::new();
 
+    if let Ok(ref orig) = orig {
+        if let Some(first_line) = orig.lines().into_iter().next() {
```

Comment:
Could this perhaps be...

```rust
if let Ok(orig) = &orig {
    for line in org.lines().take_while(|line| line.starts_with("#")) {
        out.push_str(line);
        out.push_str("\n");
    }
}
```

That way we can preserve multiline comments if necessary as well